### PR TITLE
Modernize rpcauth.py

### PR DESCRIPTION
**Pull request by sipa**
Use Python3 constructions, and f-strings.

### CODE REVIEW

**Comment by codo1** [comment]
tACK
Tested as follows: generated new userpw using an explicit password. Used it on signet through an RPC request (checked I got 'incorrect password attempt' if I changed the userpw a bit).

**Review by brunoerg — COMMENTED** [review]
from CI (locally I got same error when running `rpcauth-test`):
```py
Traceback (most recent call last):
  File "/tmp/cirrus-ci-build/ci/scratch/build/bitcoin-x86_64-w64-mingw32/src/../test/util/rpcauth-test.py", line 33, in test_generate_password
    base64.urlsafe_b64decode(password)).decode('utf-8')
  File "/usr/lib/python3.10/base64.py", line 133, in urlsafe_b64decode
    return b64decode(s)
  File "/usr/lib/python3.10/base64.py", line 87, in b64decode
    return binascii.a2b_base64(s)
binascii.Error: Incorrect padding
```

File: share/rpcauth/rpcauth.py [hunk 1]
```diff
@@ -4,22 +4,20 @@
 # file COPYING or http://www.opensource.org/licenses/mit-license.php.
 
 from argparse import ArgumentParser
-from base64 import urlsafe_b64encode
 from getpass import getpass
-from os import urandom
-
 import hmac
+import secrets
```
Comment (brunoerg) [inline share/rpcauth/rpcauth.py:9]: nit: `from secrets import token_hex, token_urlsafe`

File: share/rpcauth/rpcauth.py [hunk 2]
```diff
@@ -4,22 +4,20 @@
 # file COPYING or http://www.opensource.org/licenses/mit-license.php.
 
 from argparse import ArgumentParser
-from base64 import urlsafe_b64encode
```
Comment (brunoerg) [inline share/rpcauth/rpcauth.py:7]: Since you're replacing `base64` lib to use `secrets`, perhaps would be good to do the same in `rpcauth-test`?

**Comment by jamesob** [comment]
Github ACK https://github.com/bitcoin/bitcoin/pull/27081/commits/e4e17907b686c25dda91e569645a8f501ca48f90

**Comment by codo1** [comment]
> @codo1 If this comment can be helpful

It is. Thank you. Part of the docs you link to deserve a re-read by me.

**Pull request merged**